lib/dynflow/action/with_sub_plans.rb

```diff
@@ -2,33 +2,58 @@ module Dynflow
   module Action::WithSubPlans
```

Comment:
I wonder if the chnages on this action could be minimized, by defining some interface (methods the method needs to implement), that the middleware would call. Passing the execution tickets by modifying the events seems suboptimal. What if, instead of middleware, we would have some dependency injection here (just a method returning the semaphore-like object), that would help with the stuff? Let's talk about it on Monday